### DISCUSSION TITLE
pg_cron / seems job never executed

### DISCUSSION THREAD

**Discussion by mirus-ua** [supabase/supabase#1628] [Questions] [answered]
I tried to execute a simples CRON job with pg_cron by following the official tutorial https://supabase.io/blog/2021/03/05/postgres-as-a-cron-server and it's not working for me.

Where I'm now:
I have the pg_cron extension installed by supabase UI; I have the simplest CRON job which looks like
```
SELECT cron.schedule(
'* * * * *',
$$
  DELETE FROM data
  WHERE id=1
$$
);
```
if I execute `SELECT * FROM cron.job;` I will get it in the list of scheduled jobs.
But at the end of the day, it never will be executed.

What I already try to do:
- disable/enable extension
- create a brand new project and redo steps following the guide 

Any ideas on how I can handle this issue?

**Comment by pyoner** [comment]
I have same issue!

**Reply by mirus-ua** [reply]
Looks like pg_cron plugin doesn't have permissions to work with the public table or so.

**Reply by pyoner** [reply]
I have created the issue https://github.com/supabase/supabase/issues/1703

**Comment by mirus-ua** [comment]
https://github.com/supabase/supabase/issues/1703 here the solution

**Comment by dragarcia** [comment, marked as answer]
If anyone else gets stuck in the future, do try out this flow when using the `pg_cron` extension on the dashboard:

---

### 0. [_OPTIONAL_] Drop all scheduled jobs and the pg_cron extension itself
- This is important if you already have pg_cron enabled and have jobs set
```sql
-- repeat as much to clear all jobs
select cron.unschedule(<job id>); 

-- drop the extension
drop extension if exists pg_cron;
```
### 1. Enable pg_cron
```sql
create extension if not exists pg_cron;
```
### 2. [_IMPORTANT_] Provide the following grants to the user `postgres` (and in the future any other user that would be using the cron schema). At the baseline, it seems pg_cron uses the `postgres` user when executing its schedule.
```sql
grant usage on schema cron to postgres;
grant all privileges on all tables in schema cron to postgres;
```
### 3. Ensure that your destination table (if any) already exists beforehand.
```sql
CREATE TABLE test_table (
  id bigint GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY,
  inserted_at timestamp with time zone DEFAULT timezone('utc'::text, now()) NOT NULL,
  updated_at timestamp with time zone DEFAULT timezone('utc'::text, now()) NOT NULL,
  name text
);
```
### 4. Run your desired cron job.
```sql
select
  cron.schedule(
    'whack-every-minute', 
    '* * * * *', 
    $$
    insert into test_table(name) values ('kek');
    $$
  );
```

### 5a. Observe that the job has been successful by checking the destination table.
```sql
select * from test_table;
```
![image](https://user-images.githubusercontent.com/26374889/124260236-0a3d4e80-db62-11eb-8398-c34eb415877d.png)

### 5b. Alternatively, you can observe the `cron.job_run_details` table.
```sql
select * from cron.job_run_details;
```
![image](https://user-images.githubusercontent.com/26374889/124858920-8829ab80-dfe1-11eb-9487-57c48747e446.png)

**Reply by Musilix** [reply]
Sadly, after following these steps, it still doesn't resolve the issue. Any chance things have changed in the 3 years since this answer was posted?

**Reply by aristidesnakos** [reply]
@Musilix I restarted my project, after executing the commands above and that worked for me.
